### PR TITLE
[12.0][FIX] stock_available_unreserved

### PR DESCRIPTION
**Pull request by AdriaGForgeFlow**
This PR includes the fix proposed here #511 and also adds a pre_init_hook to ease the installation of this module in huge databases.

cc ~ @ForgeFlow

### CODE REVIEW

**Comment by pedrobaeza** [comment]
As this is done in the middle of a version, you must keep the previous computed field, although providing the new one for optimized computations.

**Comment by AdriaGForgeFlow** [comment]
> As this is done in the middle of a version, you must keep the previous computed field, although providing the new one for optimized computations.

I've brought it back. I know the computation of the field on the pre_init_hook is not exactly the one in python code but to translate the `_get_available_quantity` method to SQL language would take too long. Do you think like this is alright or I should just create the column and leave it uncomputed?

File: stock_available_unreserved/hooks.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        SELECT column_name
+        FROM information_schema.columns
+        WHERE table_name='stock_quant' AND
+        column_name='unreserved_quantity'
+    """)
+    if not cr.fetchone():
+        cr.execute(
+            """
+            ALTER TABLE stock_quant ADD COLUMN unreserved_quantity numeric
+            DEFAULT 0.0;
```
Comment (pedrobaeza) [inline stock_available_unreserved/hooks.py]: Don't add default, or it will be forever the default.

File: stock_available_unreserved/hooks.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            SELECT column_name
+            FROM information_schema.columns
+            WHERE table_name='stock_quant' AND
+            column_name='contains_unreserved'
+        """)
+    if not cr.fetchone():
+        cr.execute(
+            """
+            ALTER TABLE stock_quant ADD COLUMN contains_unreserved boolean
+            DEFAULT FALSE;
```
Comment (pedrobaeza) [inline stock_available_unreserved/hooks.py]: Don't add default

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: stock_available_unreserved/hooks.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+logger = logging.getLogger(__name__)
+
+
+def pre_init_hook(cr):
+    """
+    The objective of this hook is to speed up the installation
+    of the module on an existing Odoo instance.
+
+    Without this script, if a database has a few hundred thousand
+    partners, which is not unlikely, the update will take
```
Comment (pedrobaeza) [inline stock_available_unreserved/hooks.py]:
```suggestion
    quants, which is not unlikely, the update will take
```

File: stock_available_unreserved/tests/test_stock_available_unreserved.py [hunk 4]
```diff
@@ -214,6 +214,7 @@ def test_stock_search(self):
         no_template = self.env['product.template']
         # Start: one template with three variants.
         # All variants have zero unreserved stock
+
```
Comment (pedrobaeza) [inline stock_available_unreserved/tests/test_stock_available_unreserved.py]: Why adding an extra line here? PEP8 disregards empty lines in the middle of a method.

Comment (AdriaGForgeFlow) [reply stock_available_unreserved/tests/test_stock_available_unreserved.py]: That was the cherry-pick from jordi. I have removed it.

File: stock_available_unreserved/hooks.py (hunk 3 above)
Comment (AdriaGForgeFlow) [reply stock_available_unreserved/hooks.py]: :+1:

**Review by pedrobaeza — APPROVED** [review]
Please squash commits

**Review by marcelsavegnago — APPROVED** [review]

**Review by LoisRForgeFlow — APPROVED** [review]
LGTM

**Comment by pedrobaeza** [comment]
@AdriaGForgeFlow @LoisRForgeFlow I have just seen that you need to add a migration script calling the pre-fill of the new field, or it will take a lot also for updating the module.

**Pull request merged**